modules/justpremiumBidAdapter.js

```diff
@@ -1,274 +1,186 @@
-import bidfactory from 'src/bidfactory';
-import bidmanager from 'src/bidmanager';
-import adloader from 'src/adloader';
-import * as utils from 'src/utils';
-import adaptermanager from 'src/adaptermanager';
-
-const CONSTANTS = require('src/constants.json');
-
-const JustpremiumAdapter = function JustpremiumAdapter() {
-  const top = window.top;
-  let d;
-  let bids;
-  let cookieLoaded = false;
-  let adManagerLoaded = false;
-  let jPAM;
-  let dConfig;
-  let toLoad;
-  let server;
-
-  function isCrossOriginIframe() {
-    try {
-      return !top.document;
-    } catch (e) {
-      return true;
+import { registerBidder } from 'src/adapters/bidderFactory'
+
+const BIDDER_CODE = 'justpremium'
+const ENDPOINT_URL = top.document.location.protocol + '//pre.ads.justpremium.com/v/2.0/t/xhr'
+const pixels = [{
+  type: 'iframe',
+  src: '//us-u.openx.net/w/1.0/pd?plm=10&ph=26e53f82-d199-49df-9eca-7b350c0f9646'
+}]
+
+export const spec = {
+  code: BIDDER_CODE,
+  time: 60000,
+
+  isBidRequestValid: (bid) => {
+    return !!(bid && bid.params && bid.params.zone)
+  },
+
+  buildRequests: (validBidRequests) => {
+    const c = preparePubCond(validBidRequests)
+    const payload = {
+      zone: [...new Set(validBidRequests.map(b => {
+        return parseInt(b.params.zone)
+      }))].join(','),
+      hostname: top.document.location.hostname,
+      protocol: top.document.location.protocol.replace(':', ''),
+      sw: window.top.screen.width,
+      sh: window.top.screen.height,
+      ww: window.top.innerWidth,
+      wh: window.top.innerHeight,
```

Comment:
The above four `window.top` references will need to be in a try/catch block, or can set up a variable that access `window.top`, place that within a try/catch, and access it here